mozilla/fx-private-relay
# fix mpp-1242: dynamic purchase urls by lang & country

refactor purchase urls into new premium_subscribe_url template tag

    it accepts an accept-language value, and returns the proper subscription
    url based on the first language and/or country code in the
    accept-language header.

    it uses a hard-coded map of currency, language, and country to the
    appropriate stripe product plan ID.

## To test

1. In a browser with `en-US`, go to http://127.0.0.1:8000/accounts/profile/
   * Expected result: english text "Upgrade to ⁨Relay Premium" links contains `?plan=price_1JmRSRJNcmPzuWtRN9MG5cBy`
2. Change browser to 'de' language
3. refresh http://127.0.0.1:8000/accounts/profile/
   * Expected result: German text "Machen Sie das Upgrade auf ⁨Relay Premium" link contains `?plan=price_1JmRTDJNcmPzuWtRnJavIXXX`

Repeat for as many language & country combinations in [the spreadsheet of Relay plan IDs](https://docs.google.com/spreadsheets/d/1UhfHtQP8OOqi8tptTLjzC6dRmqSKLKqOi6Q1HO8N2ig/edit#gid=0).

## Dev note

This also introduces a new `PREMIUM_PRICE_ID_OVERRIDE` env var which restores the old `PREMIUM_PRICE_ID` UI behavior. That is, if you set `PREMIUM_PRICE_ID_OVERRIDE` env var, all purchase urls will be hard-coded to the single (USD English) product plan. This will help in local, dev, and stage environments, where we may not want to duplicate all of the Premium Relay products & plans.
